Add control plane APIs support

lxiaopei opened the pull request.

lxiaopei commented:
@mengdie-song  @annakhm   Could you help review the change? thanks a lot.

mengdie-song commented on api_cluster_control_plane.go:
```diff
@@ -0,0 +1,755 @@
+			error: localVarHttpResponse.Status,
+		}
+
+		if localVarHttpResponse.StatusCode == 201 {
```
This line will not be reached.

lxiaopei replied:
thanks, remove it

mengdie-song commented on api_cluster_control_plane.go:
```diff
@@ -0,0 +1,755 @@
+			return localVarReturnValue, localVarHttpResponse, newErr
+		}
+
+		if localVarHttpResponse.StatusCode == 400 {
```
There are some duplicate code for different status codes(400, 403 ...). They can be merged.

lxiaopei replied:
thanks, merge the code.

mengdie-song commented on api_cluster_control_plane.go:
```diff
@@ -0,0 +1,755 @@
+			return localVarReturnValue, localVarHttpResponse, newErr
+		}
+
+		if localVarHttpResponse.StatusCode == 400 {
```
Ditto.

lxiaopei replied:
thanks, merge the code.

mengdie-song commented on api_cluster_control_plane.go:
```diff
@@ -0,0 +1,755 @@
+			error: localVarHttpResponse.Status,
+		}
+
+		if localVarHttpResponse.StatusCode == 200 {
```
Ditto.

lxiaopei replied:
thanks, merge the code.

mengdie-song commented on api_client.go:
```diff
@@ -515,6 +516,7 @@ func (c *APIClient) prepareRequest(
 
 	// Generate a new request
 	if body != nil {
+		log.Printf("[DEBUG] the request to nsx-t to register method: %s, url: %s, body: %s", method, url.String(), body)
```
I am not sure if this log description is appropriate.

lxiaopei replied:
forget to remove this debug message. remove it.

annakhm commented on api_client.go:
```diff
@@ -560,18 +562,22 @@ func (c *APIClient) prepareRequest(
 		// Basic HTTP Authentication
 		if auth, ok := ctx.Value(ContextBasicAuth).(BasicAuth); ok {
 			localVarRequest.SetBasicAuth(auth.UserName, auth.Password)
+			log.Printf("[DEBUG] basic auth: %s - %s", auth.UserName, auth.Password)
```
We should not reveal sensitive info in logs, same below

lxiaopei replied:
remove the debug message.

annakhm commented:
Would you like to consider using https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt-mp for this?

annakhm commented:
Would you like to consider using https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt-mp for this?

lxiaopei commented:
> Would you like to consider using https://github.com/vmware/vsphere-automation-sdk-go/tree/master/services/nsxt-mp for this?

thanks @annakhm  @mengdie-song  for the review.  Addressed the comments. 
After this is merged , I will add the api in https://github.com/vmware/vsphere-automation-sdk-go

mengdie-song reviewed (commented):
@lxiaopei  I see there are eight commits in this PR? Could you update this?

lxiaopei commented:
> @lxiaopei I see there are eight commits in this PR? Could you update this?

merged into 1 commit.
